### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by danielsnider**

### CODE REVIEW

File: functions/DifferentialMeasurements.m [hunk 1]
```diff
@@ -114,7 +114,7 @@
                           + normalized_differences{t}.MajorAxisLength .* weights.MajorAxisLength ...
                           + normalized_differences{t}.MinorAxisLength .* weights.MinorAxisLength ...
                           + normalized_differences{t}.Solidity .* weights.Solidity ...
-                          + normalized_differences{t}.Orientation .* weights.Orientation ...
+                          + normalized_differences{t}.Orientation .* weights.Orientation;
```
Comment (danielsnider) [inline functions/DifferentialMeasurements.m]: there should be a `...` line continuation here, not a semicolin please.

File: functions/imgs_to_gif.m [hunk 2]
```diff
@@ -2,10 +2,19 @@ function images_to_gif(imgs)
...
+        else
+           imwrite(imind,cm,filename,'gif', 'DelayTime',0.4, 'WriteMode','append');
+        end
+      
+      
+%       if t==1;
+%         imwrite(imgs(:,:,t,:)./12,filename,'gif', 'DelayTime',0.5, 'Loopcount',inf);
+%       else
+%         imwrite(imgs(:,:,t,:)./12,filename,'gif', 'DelayTime',0.5, 'WriteMode','append');
+%       end
```
Comment (danielsnider) [inline functions/imgs_to_gif.m]: Please create a new file called "colour_imgs_to_gif.m" and move all this code there except the comment. Then remove the comments and "imgs_to_gif.m" will be the version that works on grey images. Thanks!

File: functions/overlay_trace_colours_on_imgs.m [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        ObjectsInFrame = CellsTable(CellsTable.Time==t,:); %ResultTable for cells in frame
+        boundaries_nuc = ObjectsInFrame.nuc_boundaries; %nuclei boundaries
+        boundaries_cyto = ObjectsInFrame.cyto_boundaries; %cytoplasm boundaries
+        NumberOfCells = size(boundaries_nuc, 1);
+        point_in_time = zeros(size(imgs,1), size(imgs,2), 3);
+
+        
+            
+            for i=1:NumberOfCells
```
Comment (danielsnider) [inline functions/overlay_trace_colours_on_imgs.m]: too much indentation

File: functions/overlay_trace_colours_on_imgs.m [hunk 4]
```diff
@@ -0,0 +1,85 @@
+function cyto_overlay = overlay_trace_colours_on_imgs(CellsTable, imgs)
+
+    labelled_by_trace_id_this = zeros(size(imgs,1), size(imgs,2), size(imgs,3), 3);
```
Comment (danielsnider) [inline functions/overlay_trace_colours_on_imgs.m]: Rename variable from "labelled_by_trace_id_this" to "labelled_by_trace" please

File: functions/overlay_trace_colours_on_imgs.m [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        boundaries_cyto = ObjectsInFrame.cyto_boundaries; %cytoplasm boundaries
+        NumberOfCells = size(boundaries_nuc, 1);
+        point_in_time = zeros(size(imgs,1), size(imgs,2), 3);
+
+        
+            
+            for i=1:NumberOfCells
+
+              Object = ObjectsInFrame(i,:);
+              trace = Object.Trace{:};
```
Comment (danielsnider) [inline functions/overlay_trace_colours_on_imgs.m]: nice code, you make it very readable

File: functions/overlay_trace_colours_on_imgs.m [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+              point_in_time(boundaries_nuc{i}) = red;
+              point_in_time(boundaries_nuc{i}+size(imgs,1)*size(imgs,2)) = green;
+              point_in_time(boundaries_nuc{i}+size(imgs,1)*size(imgs,2)*2) = blue;
+
+            end
+            
+            %fill in perimeter
+              point_in_time(:,:,1) = imfill(point_in_time(:,:,1),'holes');
+              point_in_time(:,:,2) = imfill(point_in_time(:,:,2),'holes');
+              point_in_time(:,:,3) = imfill(point_in_time(:,:,3),'holes');
```
Comment (danielsnider) [inline functions/overlay_trace_colours_on_imgs.m]: indentation problem

File: functions/overlay_trace_colours_on_imgs.m [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+%       labelled_by_size_color_fix=labelled_by_size(:,:,i);
+%       labelled_by_size_color_fix(1)=min(ResultsTable.CellSize);
+%       labelled_by_size_color_fix(2)=COLOR_LIMIT;
+%       boundries_rgb(:,:,i,:) = label2rgb(round(labelled_by_size_color_fix),'jet', 'k');
+%     end
+%     mod(sum(uint8('19oueiueoif')), 255); % calculate a colour value between 0 and 255
+%     cyto_rgb = cat(4, cyto, cyto, cyto);
+%     cyto_overlay = uint8(boundries_rgb./6) ... % segmented boundries
+%                  + uint8(cyto_rgb);           % original cyto
+%     % figure('name','cyto_overlay', 'NumberTitle','off');imshow3Dfull(uint8(cyto_overlay),[]);
```
Comment (danielsnider) [inline functions/overlay_trace_colours_on_imgs.m]: remove my comments please

File: functions/DifferentialMeasurements.m (hunk 1 above)
Comment (heathkd) [reply functions/DifferentialMeasurements.m]: Fixed

File: functions/imgs_to_gif.m (hunk 2 above)
Comment (heathkd) [reply functions/imgs_to_gif.m]: Done

File: functions/overlay_trace_colours_on_imgs.m (hunk 4 above)
Comment (heathkd) [reply functions/overlay_trace_colours_on_imgs.m]: Done

File: functions/overlay_trace_colours_on_imgs.m (hunk 3 above)
Comment (heathkd) [reply functions/overlay_trace_colours_on_imgs.m]: Fixed

File: functions/overlay_trace_colours_on_imgs.m (hunk 6 above)
Comment (heathkd) [reply functions/overlay_trace_colours_on_imgs.m]: fixed

File: functions/overlay_trace_colours_on_imgs.m (hunk 7 above)
Comment (heathkd) [reply functions/overlay_trace_colours_on_imgs.m]: Removed

File: functions/overlay_trace_colours_on_imgs.m (hunk 5 above)
Comment (heathkd) [reply functions/overlay_trace_colours_on_imgs.m]: Merci

File: functions/cell_tracking_v1_simple.m [hunk 8]
```diff
@@ -1,12 +1,13 @@
 function CellsTable = cell_tracking_v1_simple(CellsTable, composite_differences)
   %% FIND CELL TRACES
   % Initialize all trace IDs to None
+  diffTable=table(0,0,{'yo'});
```
Comment (danielsnider) [inline functions/cell_tracking_v1_simple.m]: sup!

File: functions/cell_tracking_v1_simple.m [hunk 9]
```diff
@@ -1,12 +1,13 @@
...
   CellsTable(:,{'Trace'}) = {'None'};
   % For the first frame (ie. min(CellsTable.Time) initialize the cell traces to a random UUID
   first_timepoint_cells = 1:sum(CellsTable.Time==min(CellsTable.Time));
   CellsTable.Trace(first_timepoint_cells) = uuid_array(sum(CellsTable.Time==min(CellsTable.Time)))';
   % CREATE TRACES BY FINDING CLOSEST MATCHING OBSERVATIONS FIRST BETWEEN T AND T+1
-  for timepoint=1:length(composite_differences)
+  for timepoint=1:2%length(composite_differences)
```
Comment (danielsnider) [inline functions/cell_tracking_v1_simple.m]: undo

File: functions/cell_tracking_v1_simple.m [hunk 10]
```diff
@@ -15,41 +16,51 @@
     while any(differences(:))
       % Find pair that is least different
       [current_cell_index, former_cell_index] = find(differences==min(differences(:))); % MATCH FOUND
-
+      mindiff=min(differences(:));
```
Comment (danielsnider) [inline functions/cell_tracking_v1_simple.m]: this is for debugging i guess?

File: functions/cell_tracking_v1_simple.m [hunk 11]
```diff
@@ -15,41 +16,51 @@
...
+%       min(neighbour_distances)
+%       ParentCell = PossibleParentsUpdated(find(min(neighbour_distances)),:);
+%       PossibleParentsUpdated((find(min(neighbour_distances))),:)=[]; %Heather added this 
+%       CellsTable.Trace(newborns_cells(i)) = ParentCell.Trace;
+%     end
+% 
+%     %% CELLS ENTERING FRAME
+%     % Give a trace ID to cells that were not matched
+%     cells_entering_frame = CellsTable.Time==current_timepoint & strcmp(CellsTable.Trace,'None');
+%     CellsTable.Trace(cells_entering_frame) = uuid_array(sum(cells_entering_frame));
```
Comment (danielsnider) [inline functions/cell_tracking_v1_simple.m]: uncomment

File: functions/cell_tracking_v1_simple.m [hunk 12]
```diff
@@ -15,41 +16,51 @@
...
   end
+  
+  diffTable
```
Comment (danielsnider) [inline functions/cell_tracking_v1_simple.m]: delete debugging

File: functions/isTraceIdRepeated.m [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+                trace=CellsTable.Trace(j);
+                if strcmp(currentTrace, trace)==1
+                    error = 'Trace id repeated'
+                end
+            end
+
+        end
+    end
+end
+
```
Comment (danielsnider) [inline functions/isTraceIdRepeated.m]: yes and no
yes -> good in principle
no -> slow in practice, just the law of infinitesimally small probabilities (i made that up)

**Pull request merged**